### PR TITLE
COMMERCE-12504 - Missing Data in headless-commerce-admin-catalog Endpoints

### PR DESCRIPTION
**Pull request by victorcdp**
Accommodating the changes requested by @andrea-ale-sbarra. Created new addEntity methods to be used exclusively by headless.
Used the [presence](https://github.com/liferay-commerce/liferay-portal/pull/4406/commits/b41eb74392b8d862f88c4d759d7dca5eac134231#diff-0df63818e5a102dad660cfee16d3c00c6aeeae966b0bb8057f76df31d3367294R78) of OptionValues sent to determine whether the ProductOption will import the default Option Values or not. If the user sends an array of OptionValues, then they will be created/updated; if the user sends an empty array of OptionValues, then the default ones will be imported.

[Pulled](https://github.com/liferay-commerce/liferay-portal/pull/4406/commits/5e70c5c8b25efde68b5d5cdc9a9eccb944c300d6#diff-29f705ffd258b4a5c752c2cb3c483fe05fbb49358512cf8602f966a6cbce7bd7L1102-L1108) a validation from _validateLinkedCPDefinitionOptionValueRel and put it into it's [own method](https://github.com/liferay-commerce/liferay-portal/pull/4406/commits/5e70c5c8b25efde68b5d5cdc9a9eccb944c300d6#diff-29f705ffd258b4a5c752c2cb3c483fe05fbb49358512cf8602f966a6cbce7bd7R1099-R1110) since the new add method needs this validation and the _validateLinkedCPDefinitionOptionValueRel [won't run](https://github.com/liferay-commerce/liferay-portal/pull/4406/commits/5e70c5c8b25efde68b5d5cdc9a9eccb944c300d6#diff-29f705ffd258b4a5c752c2cb3c483fe05fbb49358512cf8602f966a6cbce7bd7R1157-R1159) for new OptionValueRels.
The rest of the validations  on _validateLinkedCPDefinitionOptionValueRel don't need to be run as they're already enforced in the new addCPDefinitionOptionValueRel method.

Relevant Issue: [COMMERCE-12504](https://liferay.atlassian.net/browse/COMMERCE-12504)
Resent of #4387 and #4375.

### CODE REVIEW

**Comment by victorcdp** [comment]
ci:test:sf

**Comment by victorcdp** [comment]
ci:test:relevant

**Comment by victorcdp** [comment]
ci:test:commerce-functional

**Comment by victorcdp** [comment]
ci:test:relevant

**Review by andrea-ale-sbarra — CHANGES_REQUESTED** [review]
Please get rid of unused method too. I saw that we have in the code base an add method for CPDefinitionOptionRel that is never used.
Thank you!

File: modules/apps/commerce/headless/headless-commerce/headless-commerce-admin-catalog-impl/src/main/java/com/liferay/headless/commerce/admin/catalog/internal/jaxrs/exception/mapper/CPDefinitionOptionValueRelSkuExceptionMapper.java [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+import org.osgi.service.component.annotations.Component;
+
+/**
+ * @author Joao Cordeiro
+ */
+@Component(
+	property = {
+		"osgi.jaxrs.application.select=(osgi.jaxrs.name=Liferay.Headless.Commerce.Admin.Catalog)",
+		"osgi.jaxrs.extension=true",
+		"osgi.jaxrs.name=Liferay.Headless.Commerce.Admin.Catalog.OptionValueSkuIdExceptionMapper"
```
Comment (andrea-ale-sbarra) [inline modules/apps/commerce/headless/headless-commerce/headless-commerce-admin-catalog-impl/src/main/java/com/liferay/headless/commerce/admin/catalog/internal/jaxrs/exception/mapper/CPDefinitionOptionValueRelSkuExceptionMapper.java:24]: rename the class too -> OptionValueSkuIdExceptionMapper

**Review by andrea-ale-sbarra — CHANGES_REQUESTED** [review]
Overall looks good.
Justa class rename.
Thank you!

File: modules/apps/commerce/headless/headless-commerce/headless-commerce-admin-catalog-impl/src/main/java/com/liferay/headless/commerce/admin/catalog/internal/jaxrs/exception/mapper/CPDefinitionOptionValueRelSkuExceptionMapper.java (hunk 1 above)
Comment (victorcdp) [reply modules/apps/commerce/headless/headless-commerce/headless-commerce-admin-catalog-impl/src/main/java/com/liferay/headless/commerce/admin/catalog/internal/jaxrs/exception/mapper/CPDefinitionOptionValueRelSkuExceptionMapper.java]: Done. Thank you!

**Comment by victorcdp** [comment]
ci:test:sf

**Comment by victorcdp** [comment]
@andrea-ale-sbarra Added [preselected](https://github.com/liferay-commerce/liferay-portal/pull/4406/commits/b5b3c74d1f03df4eac2f5ddf9055ebe8bb2de6c5#diff-29f705ffd258b4a5c752c2cb3c483fe05fbb49358512cf8602f966a6cbce7bd7R121) to CPDefinitionOptionValueRel as well since i had forgotten, and [updated](https://github.com/liferay-commerce/liferay-portal/pull/4406/commits/c85a652bf8f4f972a54b6743c5c1c83dc7500028#diff-9e2c46ea5400e2eb056d2903af5006979ea6bee22c47afd6899a2b84ff8be417L1045-R1153) the priceType validation since the `!cpDefinitionOptionRel.isPriceContributor()` line accepted everything, even if it wasn't "static", "dynamic", or "".

**Review by andrea-ale-sbarra — APPROVED** [review]
LGTM.
Thank you!

**Comment by andrea-ale-sbarra** [comment]
ci:test:sf

**Comment by andrea-ale-sbarra** [comment]
ci:test:relevant

**Comment by andrea-ale-sbarra** [comment]
ci:test:commerce-functional

**Comment by victorcdp** [comment]
@andrea-ale-sbarra updated the [validatePriceType](https://github.com/liferay-commerce/liferay-portal/pull/4406/commits/1030bfdcf6f326ba7e31c247856baa52db30345a#diff-9e2c46ea5400e2eb056d2903af5006979ea6bee22c47afd6899a2b84ff8be417R1138-R1146) again since it was causing issues. This time i just validate the allowed priceTypes at the beginning of the method

**Comment by victorcdp** [comment]
ci:test:sf

**Comment by victorcdp** [comment]
ci:test:relevant

**Comment by victorcdp** [comment]
ci:test:commerce-functional

**Comment by andrea-ale-sbarra** [comment]
ci:stop

**Comment by andrea-ale-sbarra** [comment]
resent here https://github.com/liferay-commerce/liferay-portal/pull/4434

**Pull request closed without merging**